darwin/entry.m

```diff
@@ -155,6 +171,31 @@ void uiEntryOnChanged(uiEntry *e, void (*f)(uiEntry *, void *), void *data)
 	e->onChangedData = data;
 }
 
+// NOTE: for search widgets on OSX, setting OnFinished() alters the behaviour
+// (by setting sendsWholeSearchString).
+// Standard text and password entry widgets are not affected.
+// background:
+// On OSX, there doesn't seem to be any simple way to catch
+// both 'changed' events and 'enter' (finished editing) events on
+// search widgets. Instead, there just a single 'search' event, and flags
+// to determine when it triggers. By default, it triggers after each keypress
+// (with a little delay in case the user is still typing).
+// There's also an option to change the behaviour to trigger only when the
+// enter key is hit, or the search icon is pressed. This is the
+// sendsWholeSearchString flag, and we'll set it if (and only if) OnFinished()
+// is used.
+void uiEntryOnFinished(uiEntry *e, void (*f)(uiEntry *, void *), void *data)
+{
+	e->onFinished = f;
+	e->onFinishedData = data;
+	if (isSearchField(e->textfield)) {
+		NSSearchField *s;
+		s = (NSSearchField *) (e->textfield);
+		// TODO requires OSX >= 10.10  (is that an issue?)
```

Comment:
Calling it directly is, but this kind of thing is what `future.m` is for.

Reply:
Actually the equivalent property on NSSearchFieldCell is 10.3 and newer, so just replace `s` in the call with `[s cell]` (or preferably, given the coding style I've written myself into with this and that I should probably clan up someday, a cast version of `[s cell]`).